src/Core/Aetherling/Passes/ThroughputModifications.hs

```diff
@@ -1,339 +1,508 @@
+
+-- LEAF, MODIFIABLE RATE
+-- If possible, speed up by changing the rate. Otherwise, just try to return
+-- an op with the same or greater throughput than the original.
+-- However, behavior is undefined if requesting an invalid throughput multiplier
+-- that doesn't match the divisibility requirements such as those for linebuffer.
+-- Not going to speed these up by wrapping them in a map because that would
+-- cause inappropriate behavior in the linebuffer. The linebuffer has complex
+-- internal state and mapping over it will cause it to behave unintuitively.
+-- For example, an LineBuffer [2] [3] [300] T_Int will reading in 2 int per clock
+-- and emit two overlapping windows of consecutive ints.
+-- A Map 2 (LineBuffer [1] [3] [300] T_Int) will emit two windows, but the
+-- ints in the windows will not be consecutive. The first window will have ints
+-- that came in 0, 2, and 4 (0-indexed) and the second window will have ints
+-- indexed 1, 3, and 5.
+
+-- speed up inner most dimension first. Once you have an entire row input per
+-- clock, then speed up the outer dimensions like number of rows per clock.
+-- see increaseLBPxPerClock for more information. 
+-- NOTE: increaseLBPxPerClock expects dimensions from innermost to outermost
+-- and LineBuffer stores them in the opposite order. Thus, they must be
+-- reversed when calling the helper function.
+attemptSpeedUp requestedMult (LineBuffer p w img t bc) =
   (LineBuffer (reverse reversedNewP) w img t bc, actualMult)
   where
     (reversedNewP, actualMult) =
-      increaseLBPxPerClock (reverse p) (reverse img) throughMult
--- don't increase the arr size, duplicate it so it can be fed to other maps
-attemptSpeedUp throughMult op@(Constant_Int _) = (MapOp throughMult op, throughMult)
-attemptSpeedUp throughMult op@(Constant_Bit _) = (MapOp throughMult op, throughMult)
--- not going to change SLen in as, if 1, don't want to cause it to become
--- fractional
-attemptSpeedUp throughMult (SequenceArrayRepack (sLenIn, oldArrLenIn)
+      increaseLBPxPerClock (reverse p) (reverse img) requestedMult
+
+-- could decrease sLenIn and sLenOut as increase throughput, but not going
+-- to do that as don't want to deal with fractional sequence lengths,
+-- which could happen if dividing sLenIn or sLenOut 
+attemptSpeedUp requestedMult (SequenceArrayRepack (sLenIn, oldArrLenIn)
                               (sLenOut, oldArrLenOut) t) =
-  (SequenceArrayRepack (sLenIn, oldArrLenIn * throughMult)
-    (sLenOut, oldArrLenOut * throughMult) t, throughMult)
-attemptSpeedUp throughMult op@(ArrayReshape _ _) =
-  (MapOp throughMult op, throughMult)
-attemptSpeedUp throughMult op@(DuplicateOutputs _ _) =
-  (MapOp throughMult op, throughMult)
-
--- is this an ok inverse of our decision to only underutil for slowdown of seq operators?
--- should this try to speed up children before itself?
-attemptSpeedUp throughMult (MapOp par innerOp) | isComb innerOp =
```

Comment:
Would be nice to comment the intuitive description here.  Speeding up a map by multiplier M is just multiplying the width parameter of the map.

Reply:
We should also note that the current implementation does not try to recursively speedup the child node. It just parallelizes in a data parallel (input token parallel) way.

Reply:
1. This is deleted code not in the code following the commit. Were you refering to the new implementation at https://github.com/David-Durst/aetherlingHaskellIR/pull/14/files/76c0a4810fb37bdc063954ba40df55af21cf309b#diff-a4e9be19161290150b0e024cdc796b07R190
2. I do give the intuitive explanation at https://github.com/David-Durst/aetherlingHaskellIR/pull/14/files/76c0a4810fb37bdc063954ba40df55af21cf309b#diff-a4e9be19161290150b0e024cdc796b07R181

Reply:
I also note the speeding up of the rate before speeding up the children at https://github.com/David-Durst/aetherlingHaskellIR/pull/14/files/76c0a4810fb37bdc063954ba40df55af21cf309b#diff-a4e9be19161290150b0e024cdc796b07R44. 

The philosophy for these types of nodes is not to totally ignore the child nodes, but to speed them up only if you can't speed up the parent. Since you can always speed up the map op by multiplying its parameter. the map's children are never speed up.